src/hooks/useMultiplayerState.ts

```diff
@@ -61,76 +61,76 @@ export function useMultiplayerState(roomId: string) {
     undoManager.redo();
   }, []);
 
-  /**
-   * Callback to update user's (self) presence
-   */
   const onChangePresence = useCallback((app: TldrawApp, user: TDUser) => {
-    if (!app.room) return;
-    room.setPresence({ id: app.room.userId, tdUser: user });
+    if (app.room) {
+      room.setPresence({ id: app.room.userId, tdUser: user });
+    }
   }, []);
 
-  /**
-   * Update app users whenever there is a change in the room users
-   */
   useEffect(() => {
-    if (!app || !room) return;
-
-    const unsubOthers = room.subscribe("others", (users) => {
-      if (!app.room) return;
-
-      const ids = users
-        .filter((user) => user.presence && user.presence.tdUser)
-        .map((user) => user.presence!.tdUser!.id);
-
-      // remove any user that is not connected in the room
-      Object.values(app.room.users).forEach((user) => {
-        if (user && !ids.includes(user.id) && user.id !== app.room?.userId) {
-          app.removeUser(user.id);
+    if (app && room) {
+      const unsubOthers = room.subscribe("others", (users) => {
+        if (app.room) {
+          const ids = users
+            .filter((user) => user.presence && user.presence.tdUser)
+            .map((user) => user.presence!.tdUser!.id);
+
+          Object.values(app.room.users).forEach((user) => {
+            if (
+              user &&
+              !ids.includes(user.id) &&
+              user.id !== app.room?.userId
+            ) {
+              app.removeUser(user.id);
+            }
+          });
+
+          app.updateUsers(
+            users
+              .filter((user) => user.presence && user.presence.tdUser)
+              .map((other) => other.presence!.tdUser!)
+              .filter(Boolean)
+          );
         }
       });
 
-      app.updateUsers(
-        users
-          .filter((user) => user.presence && user.presence.tdUser)
-          .map((other) => other.presence!.tdUser!)
-          .filter(Boolean)
-      );
-    });
-
-    return () => {
-      unsubOthers();
-    };
-  }, [app]);
+      return () => {
+        unsubOthers();
+      };
+    }
+  }, [app, room]);
 
   useEffect(() => {
-    if (!app) return;
-
-    function handleDisconnect() {
-      provider.disconnect();
-    }
+    if (app) {
```

Comment:
Why have you switched to this version from
```ts
if (!app) return;
```
? I would suggest to go back to the old version, it's better to keep nested code blocks number as low as possible, especially if there is no other logic for the _else_ like in this case.